contracts/Create2Transfer.sol

```diff
@@ -86,11 +86,12 @@ contract Create2Transfer {
         bytes32 stateRoot,
         bytes memory txs,
         Types.StateMerkleProof[] memory proofs,
-        uint256 tokenType,
         uint256 feeReceiver
     ) public pure returns (bytes32, Types.Result result) {
         uint256 length = txs.create2Transfer_size();
         uint256 fees = 0;
+        // tokenType should be the same for all states in this commit
+        uint256 tokenType = proofs[0].state.tokenType;
```

Comment:
We are taking the first tokenType provided by the disputer. In the first validateSender the tokenType will trivially match. This tokenType can be trusted if it passed the merkle inclusion check in the first validateSender. 

Then in the later processTx, this tokenType can use to verify if all of the states of sernders, receivers, and fee receivers, have the same tokenType.

Reply:
Yeah that makes a lot of sense